src/sagemaker/jumpstart/utils.py

```diff
@@ -732,18 +738,33 @@ def resolve_estimator_sagemaker_config_field(
     return field_val
 
 
-def is_valid_model_id(
+def validate_model_id_and_get_type(
     model_id: Optional[str],
     region: Optional[str] = None,
     model_version: Optional[str] = None,
     script: enums.JumpStartScriptScope = enums.JumpStartScriptScope.INFERENCE,
     sagemaker_session: Optional[Session] = constants.DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
-) -> bool:
-    """Returns True if the model ID is supported for the given script.
+) -> enums.JumpStartModelType:
+    """Returns model type if the model ID is supported for the given script.
 
     Raises:
         ValueError: If the script is not supported by JumpStart.
     """
+    def _get_model_type(
+        model_id: str,
+        open_source_models: List[str],
```

Comment:
It would be nice for these to be sets for easier O(1) lookup rather than O(n)

Reply:
Yes this is misleading typing, they are sets already. Updated